script.go

```diff
@@ -34,6 +34,12 @@ type Pipe struct {
 	// because pipe stages are concurrent, protect 'err'
 	mu  *sync.Mutex
 	err error
+
+	// env contains the environment to run any exec commands with.
+	// Each entry in the array should be of the form key=value.
+	// If env is not nil, it will replace the default environment variables
+	// when executing commands.
+	env []string
```

Comment:
Is this protected by the mutex? Does it need to be?

Reply:
this should be protected by the mutex based on the race condition that https://github.com/bitfield/script/pull/209 fixes! i'll make the change